packages/core/src/connectors/casperSigner.ts

```diff
@@ -41,17 +41,19 @@ CasperSignerConnectorOptions
       },
       ...defaultOptions,
     };
-
     super({ options });
+
+    const provider = options.getProvider();
+    this.isReady = !!provider;
   }
 
-  // eslint-disable-next-line @typescript-eslint/require-await
-  public async getProvider(): Promise<CasperLabWindowGlobal> {
+  public getProvider(): CasperLabWindowGlobal {
```

Comment:
If there's no specific logic for each connector, we can consider move common function like `getProvider` to `Connector` base class